### ISSUE TITLE
missing folder release/Magarena/decks/firemind leads to exceptions

### ISSUE THREAD

**Issue by firemind** [magarena/magarena#1051]
I stumbled upon this while trying to update my workers to 1.81

If a deck is selected that is not pre-built DeckUtils.getDeckType will try to access getFiremindDecksFolder() which is release/Magarena/decks/firemind and is not distributed with the release. So if a player (or the duel runner) didn't download firemind decks first this will lead to an exception:

java.nio.file.NoSuchFileException: /home/mike/git/magarena/release/Magarena/decks/firemind

A simple fix for this is adjusting the .gitignore file and adding the folder https://github.com/firemind/magarena-csm-auto/commit/2b7b9880baca51d18b76a43175bd02a514abeb57

However the release still breaks the ability to run firemind duels since the worker so far saved the decks to a separate duels folder. I adjusted the path to one that is of a known deck type in this commit https://github.com/firemind/magarena-csm-auto/commit/6bf61e5b907cac30a1b7166f0791d848e1fe2fc6

Is that the correct place to put them?

**Comment by lodici** [comment]
> If a deck is selected that is not pre-built DeckUtils.getDeckType will try to access getFiremindDecksFolder()...

Sorry, I am not sure what this means or how to replicate this issue from the UI but if it is a case of ensuring the ``decks\firemind`` directory always exists then you could update ``getFiremindDecksFolder()`` in ``DeckUtils`` to 
```
public static Path getFiremindDecksFolder() {
    Path decksPath = Paths.get(getDeckFolder()).resolve("firemind");
    MagicFileSystem.verifyDirectoryPath(decksPath);
    return decksPath;
}
```

**Comment by firemind** [comment]
I was able to replicate it with the UI by simply deleting the firemind folder (which would be the case with a fresh install)
`rm -r release/Magarena/decks/firemind/`

It doesn't crash the UI but it does continuously log to stdout when I start a game with two player decks. I started magarena with release/Magarena.sh

I just tested it and your proposed solution works too. Not sure which is better (including the folder in the release or always checking if it exists).

 Could you verify that my solution for the second problem is the correct way to go so I can create a PR? https://github.com/firemind/magarena-csm-auto/commit/6bf61e5b907cac30a1b7166f0791d848e1fe2fc6

**Comment by lodici** [comment]
Actually I was able to generate an exception by switching off my internet connection, installing a new version of Magarena, then attempting to access Firemind top decks from the deck selection screen. Also with internet connection on and after downloading top decks for the day, if I then delete the firemind deck folder selecting top decks will cause an exception.

I have committed a fix that basically ensures the firemind folder is created if it does not exist. Regarding your changes to ``FiremindDuelRunner`` it's your baby but I see nothing that adversely affects the rest of Magarena so PR away.
